tests/cases/compiler/implicitIndexSignatures.ts

```diff
@@ -43,3 +43,15 @@ function f4() {
     const v3 = getNumberIndexValue(o1);
     const v4 = getNumberIndexValue(o2);
 }
+
+function f5() {
+    enum E1 { A, B }
+    enum E2 { A = "A", B = "B" }
+    enum E3 { A = 0, B = "B" }
+    const v1 = getStringIndexValue(E1);
+    const v2 = getStringIndexValue(E2);
+    const v3 = getStringIndexValue(E3);
+    const v4 = getNumberIndexValue(E1);
+    const v5 = getNumberIndexValue(E2);

```

Comment:
Why is this `unknown` and not an error? (Is it just because inference fails and "a thing with an inferrable index" is assignable to a `{[x: number]: unknown}`?) I'd think that yeah, `E2` has an inferrable index type, sure, but we know that it has no numbers in it at all. But, from the stance that the number index signature (were it present in the type) would need to be assignable to the string index, this'd need to be `E2`.

Reply:
Because we infer nothing for `T` in the call to `getNumberIndexValue`. Keep in mind we're not doing an assignment here, we're just doing inference.

Reply:
Well, we are doing an assignment - we're saying that a `{[x: string]: E2}` is assignable to a `{[x: number]: unknown}`. I guess the output type here just irks me a tad - it could only possibly be an `E2` or `keyof typeof E2`, rather than an `unknown`.

Like, if I write `enum StrNum { Zero = "0", One = "1" }`, indexing by `0` or `1` is gunna get me `Zero` or `One`, likewise if I write `enum NumStr { "0" = "Zero", "1" = "One" }`, indexing by `Zero` or `One` is gunna get me `"0"` or `"1"`.

Reply:
No, indexing your `StrNum` by `0` or `1` is going to get you `undefined` (there's no reverse mapping for string valued members), and `NumStr` is an error because members can't have numeric names.

Reply:
Anyway, the `unknown` type is consistent with what we infer for object literal types that have no numerically named members. In some sense it would be more consistent to return `undefined` since that's what you're going to get, but that really is an orthogonal issue.

Reply:
> there's no reverse mapping for string valued member

I always forget about that.

> Anyway, the unknown type is consistent with what we infer for object literal types that have no numerically named members

Oh, ick. Ok.